DatadogCore/Tests/Datadog/InternalProxyTests.swift

```diff
@@ -11,7 +11,7 @@ import DatadogInternal
 @testable import DatadogCore
 
 class InternalProxyTests: XCTestCase {
-    let telemetry = TelemetryMock()
+    let telemetry = TelemetryReceiverMock()
```

Comment:
what's the difference here? 🤔

Reply:
The `TelemetryMock` was conforming to two distinct protocols: `Telemetry` and `FeatureMessageReceiver`. Both have different purpose, so it didn't make sense to have a single mock implementation. I made a split:
* `TelemetryMock: Telemetry`
* `TelemetryReceiverMock: FeatureMessageReceiver`

With no split, I almost ended up re-implementing this mock because it wasn't obvious at first glance how to spy core telemetry in tests.